test/cql-pytest/test_compaction_tombstone_gc.py

```diff
@@ -0,0 +1,141 @@
+            assert rows == {2}, f"Expected rows=={2} but got {rows}: res={list(res)}"
+
+def test_tombstone_gc_with_delete_in_memtable(scylla_only, cql, test_keyspace):
+    """
+    Reproduce https://github.com/scylladb/scylladb/issues/20423
+    """
+    schema = "k int, v int, primary key (k, v)"
+    with new_test_table(cql, test_keyspace, schema, extra="with gc_grace_seconds = 0") as table:
+        with nodetool.no_autocompaction_context(cql, table):
+            # Insert initial data into the base table
+            cql.execute(f"insert into {table} (k, v) values (1, 1) using timestamp 1 and ttl 1")
+            cql.execute(f"insert into {table} (k, v) values (1, 2) using timestamp 2")
+            cql.execute(f"insert into {table} (k, v) values (1, 3) using timestamp 3")
+            nodetool.flush(cql, table)
+            cql.execute(f"delete from {table} using timestamp 4 where k=1 and v=2")
+            # Flush all tables explicitly now, since this is skipped in the test on purpose by next major compaction.
+            nodetool.flush_all(cql)
+
+            # let gc_grace_seconds and ttl pass
+            time.sleep(2)
+            # Insert backdated delete into the memtable.  It should not inhibit tombstone_gc
+            cql.execute(f"delete from {table} using timestamp 2 where k=1 and v=3")
+            # The following insert should not inhibit tombstone_gc since its timestamp is greater than the tombstones
+            cql.execute(f"insert into {table} (k, v) values (1, 4) using timestamp 5")
+            # do not flush before major compaction
+            nodetool.compact(cql, table, flush_memtables=False)
+
+            res = cql.execute(f"select * from mutation_fragments({table})")
+            sstables = set()
+            keys = set()
+            rows = set()
+            for r in res:
+                if "sstable" in r.mutation_source:
+                    print(r)
+                    if r.mutation_fragment_kind == "partition start":
+                        sstables.add(r.mutation_source)
+                        keys.add(r.k)
+                    elif r.mutation_fragment_kind == "clustering row":
+                        rows.add(r.v)
+                        assert not "tombstone" in r.metadata
+
+            assert len(sstables) == 1, f"Expected single sstable but saw {len(sstables)}: res={list(res)}"
+            assert keys == {1}, f"Expected keys=={1} but got {keys}: res={list(res)}"
+            assert rows == {3}, f"Expected rows=={3} but got {rows}: res={list(res)}"
+
+def test_tombstone_gc_with_materialized_view_update_in_memtable(scylla_only, cql, test_keyspace):
+    """
+    Reproduce https://github.com/scylladb/scylladb/issues/20424
+    """
+    schema = "k int primary key, v int, w int"
+    with new_test_table(cql, test_keyspace, schema) as table:
+        # Create a materialized view with same partition key as the base, and using a static column in the base as a clustering key in the view
```

Comment:
The comment's mentioning a "static column" is wrong - it's a regular column, not a static column (the phrase "static column" has a very specific meaning in CQL, and v isn't one).

Reply:
I found it herd to reproduce the issue with regular columns that are not static.
If you have a reproducer I'll be happy to add it.

Reply:
you're right that the static column in the base becomes a regular column in the view, is that what you meant?

Reply:
I don't understand - I didn't see the keyword "static" in the schema. What am i missing?

Reply:
s/static/regular/ in the next version

Reply:
> I don't understand - I didn't see the keyword "static" in the schema. What am i missing?

There is no clustering key in the base-table schema.

Reply:
> s/static/regular/ in the next version

Good, this is exactly what I asked. You kept repeating this word "static" in the comments and in your answers, but it was never a static column (which is a very specific CQL feature which you didn't use), it was a regular column. I agree that in a table with no clustering keys, every column is similar to a "static" column, but it's still a different concept and it really confused me - I felt like I must have missed something.